src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -100,7 +100,15 @@ public void Sort()
         {
             if (!MaintainTestOrder)
             {
-                this.tests.Sort();
+                this.tests.Sort(delegate (ITest x, ITest y)
+                {
+                    var xKey = 0;
+                    var yKey = 0;
+                    if (x.Properties.ContainsKey("order")) int.TryParse(x.Properties["order"][0].ToString(), out xKey);
+                    if (y.Properties.ContainsKey("order")) int.TryParse(y.Properties["order"][0].ToString(), out yKey);
+                    return xKey.CompareTo(yKey) != 0
+                    ? xKey.CompareTo(yKey) : x.Name.CompareTo(y.Name); ;
```

Comment:
My 2 cents.. I recommend this should be retained, because in the previous issue comments someone has listed a work around as using A_ B_ etc for the test name, so you may well have some customers out there that are using this, so if you change the order it may break someones test when they upgrade. But it's up to you mate :)